### ISSUE TITLE
Error while restoring the dependencies of .NET MAUI App on .NET 8

### ISSUE THREAD

**Issue by egvijayanand** [dotnet/maui#13492]
### Description

Error while restoring the dependencies of .NET MAUI App on .NET 8

E:\MAUI\NET_8\Preview_1\MauiApp1\MauiApp1.csproj : error NU1101: **Unable to find package Microsoft.NET.ILLink.Tasks**. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, nuget.org [E:\MAUI\NET_8\Preview_1\MauiApp1\MauiApp1.sln]

### Steps to Reproduce

1. Install .NET 8 Preview 1 SDK
    - Get the installer from [here](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
2. Install .NET MAUI SDK
    - dotnet workload install maui
3. Create a new .NET MAUI App from CLI
    - dotnet new maui -o MauiApp1
4. Restore the dependencies
    - dotnet restore 

### Link to public reproduction project repository

Default template itself is suffice, no authored code

### Version with bug

8.0 previews

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

All platforms

### Did you find any workaround?

Nope

### Relevant log output

```shell
E:\MAUI\NET_8\Preview_1\MauiApp1\MauiApp1.csproj : error NU1101: Unable to find package Microsoft.NET.ILLink.Tasks. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, nuget.org [E:\MAUI\NET_8\Preview_1\MauiApp1\MauiApp1.sln]
```

**Comment by Eilon** [comment]
@Redth - is .NET 8 preview 1 supposed to work with MAUI? I remember something mentioned earlier today but I don't recall the details.

**Comment by egvijayanand** [comment]
Yes supported as per this [link](https://github.com/dotnet/core/blob/main/release-notes/8.0/preview/8.0.0-preview.1.md).

And getting errors only when iOS/MacCatalyst is part of the target platforms. Restore works fine with Android/Windows as their target platforms. Haven't tried Tizen.

**Comment by egvijayanand** [comment]
Relevant issue:

https://github.com/dotnet/runtime/issues/82241

**Comment by agocke** [comment]
Package was missing on NuGet. It's been uploaded now -- this should be fixed.

**Comment by egvijayanand** [comment]
It's working now. Marking the issue as resolved.
